client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -294,6 +293,19 @@ public final TasksClient tasks() {
         return tasksClient;
     }
 
+    /**
+     * A wrapper for the {@link RestHighLevelClient} that provides methods for
+     * accessing the Elastic Licensed X-Pack APIs that are shipped with the
+     * default distribution of Elasticsearch. All of these APIs will 404 if run
+     * against the OSS distribution of Elasticsearch.
+     * <p>
+     * See the <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/xpack-api.html">
+     * X-Pack APIs on elastic.co</a> for more information.
+     */
+    public final XPackClient xPack() {
```

Comment:
I think that the casing should be `xpack`  here and probably everywhere else (xpackClient etc.)

Reply:
I started that way but went to this because it felt more right. I'm not sure why.....

Reply:
[This](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case) is what I kind of think about in my head for these things.

Reply:
I suppose that spits out `Xpack` instead of `XPack`. I'll change it.

Reply:
I don't have an opinion, I only looked at the existing spec and how naming is defined there so that the test that we've been working on in #31825 succeeds.